caddy/caddy.go

```diff
@@ -243,6 +249,186 @@ func parseCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, error)
 	return m, err
 }
 
+func parseFrankenPHP(h httpcaddyfile.Helper) ([]httpcaddyfile.ConfigValue, error) {
+
+	if !h.Next() {
+		return nil, h.ArgErr()
+	}
+
+	frankenphpTransport := FrankenPHPModule{}
+	extensions := []string{".php"}
+	indexFile := "index.php"
+	tryFiles := []string{}
+
+	userMatcherSet, err := h.ExtractMatcherSet()
+
+	if err != nil {
+		return nil, err
+	}
+
+	dispenser := h.NewFromNextSegment()
+
+	for dispenser.Next() {
+		for dispenser.NextBlock(0) {
+			if dispenser.Nesting() != 1 {
+				continue
+			}
+
+			switch dispenser.Val() {
+			case "root":
+				if !dispenser.NextArg() {
+					return nil, dispenser.ArgErr()
+				}
+				frankenphpTransport.Root = dispenser.Val()
+				dispenser.Delete()
+				dispenser.Delete()
+
+			case "env":
+				args := dispenser.RemainingArgs()
+				dispenser.Delete()
+				for range args {
+					dispenser.Delete()
+				}
+				if len(args) != 2 {
+					return nil, dispenser.ArgErr()
+				}
+				if frankenphpTransport.Env == nil {
+					frankenphpTransport.Env = make(map[string]string)
```

Comment:
You could probably initialize the map outside of the loop to simplify the code and remove this `if`.